sys/netpfil/pf/pf.c

```diff
@@ -576,8 +576,10 @@ static bool
 pf_is_loopback(sa_family_t af, struct pf_addr *addr)
 {
 	switch (af) {
+#ifdef INET
 	case AF_INET:
 		return IN_LOOPBACK(ntohl(addr->v4.s_addr));
+#endif
 	case AF_INET6:
```

Comment:
Wouldn't this case also need an `#ifdef INET6`?

Reply:
no, because `IN6_IS_ADDR_LOOPBACK()` doesn't depend on any symbols.

`IN_LOOPBACK()` is strange because of the sysctl `net.inet.ip.loopback_prefixlen` which is why it references a VNET symbol.  

this is not the first time i've fixed this exact same bug -- i'm wondering if we should just remove this 'feature' entirely.

Reply:
I see, in efe58855f3ea2cfc24cb705aabce3bc0fe1fb6d5, implementing [draft-schoen-intarea-unicast-127-06](https://datatracker.ietf.org/doc/draft-schoen-intarea-unicast-127/). Thank you for the info.

Reply:
i wasn't aware of that I-D, but i accept this as a reason not to remove the feature :-)